src/gui/clips/PatternClipView.cpp

```diff
@@ -115,12 +117,15 @@ void PatternClipView::paintEvent(QPaintEvent*)
 	
 	// bar lines
 	const int lineSize = 3;
+	bar_t t = Engine::patternStore()->lengthOfPattern(m_patternClip->patternIndex());
+	int offset = fmod(static_cast<double>(m_patternClip->startTimeOffset()) / TimePos::ticksPerBar(), t)
+			* pixelsPerBar();
```

Comment:
Just one more floating point calculation here that could be converted to use integers like the others.

Reply:
I actually kept it because I did not introduce variable t. It is used a few lines later in the loop together with pixelsPerBar(). I also need to use pixelsPerBar() for conversion of the startTimeOffset to pixels. To work with ticks instead (and therefore not divide) would require either of two options:
a) dividing pixelsPerBar by ticksPerBar to then multiply with ticks - this basically shifts the problem; or
b) writing a second pixelsPerTick member which internally does not rely on division - otherwise it would just shift the problem again.
I don't know if it is possible to not use any division in this case without going deeper into code (float m_ppb and stuff).
Please tell me, if I missed a simple solution here.

Reply:
I could drop the use fmod, though: `int offset = (m_patternClip->startTimeOffset() % (t * TimePos::ticksPerBar())) * (pixelsPerBar() / TimePos::ticksPerBar());`.
Would that be more preferable?

Reply:
I will drop the use of fmod though and create a pixelsPerPattern variable to boost the loop.

Reply:
That makes sense - there's no need to overcomplicate things here, so the changes look good as you now have them.